public/catalog/kitchensink.yaml

```diff
@@ -104,4 +106,32 @@ nodes:
   label: Node 3
   description: Description 3
   category: grouptests
+- title: Node to test pruning of defaultstest
+  id: defaultprune
+  label: Default prune?
+  category: defaultstest
+  uiSchema:
+    prop4:
+      items:
+        ui:widget: radio
+  schema:
+    type: object
+    additionalProperties: false
+    properties:
+      prop1:
+        title: String with no default
+        type: string
+      prop2:
+        title: String with default empty string
+        type: string
+        default: ''
+      prop3:
+        title: String with default x string
+        type: string
+        default: 'x'
```

Comment:
If I replace this with an empty string and press submit, it is automatically reset to 'x'. Is this intended behaviour?

Reply:
Same behaviour happens for list or array items with a default value. It is not possible to replace a default string by an empty string

Reply:
Having a non empty default string mostly happens when you also have an enum. So I think it is fine.